Made changes to resize the heap incase Alloc fails

gautham-bhat-k opened the pull request.

gautham-bhat-k committed:
Made changes to resize the heap incase Alloc fails

sanish81 commented on src/kvs_radix_tree.cc:
```diff
@@ -139,8 +139,13 @@ int KVSRadixTree::Put (char const *key, size_t const key_len,
+		heap_->Resize(2*size);
+		val_gptr = heap_->Alloc(op, val_len+sizeof(ValBuf));
+		if (!val_gptr.IsValid())
+        	return -1;
```
The "return" statement can be tab-indented to the right.

sanish81 commented on src/kvs_radix_tree.cc:
```diff
@@ -139,8 +139,13 @@ int KVSRadixTree::Put (char const *key, size_t const key_len,
-        return -1;
+    if (!val_gptr.IsValid()) {
+		size_t size = heap_->Size();
+		heap_->Resize(2*size);
```
Error return can be checked everytime Resize() is called.

sanish81 commented on src/radix_tree.cc:
```diff
@@ -76,7 +76,12 @@ RadixTree::RadixTree(Mmgr *Mmgr, Heap *Heap, RadixTreeMetrics* Metrics, Gptr Roo
     if (root==0) {
         root = heap->Alloc(sizeof(Node));
-        assert(root.IsValid());
+		if(!root.IsValid()){
```
alignment

sanish81 commented on src/radix_tree.cc:
```diff
@@ -328,7 +333,12 @@ TagGptr RadixTree::put(const char *key, const size_t key_size, Gptr value, Updat
                 while(new_leaf_ptr==0 && (cnt--)>0)
                     new_leaf_ptr = heap->Alloc(sizeof(Node));
-                assert(new_leaf_ptr.IsValid());
+				if(!new_leaf_ptr.IsValid()){
```
alignment issues.

sanish81 commented on src/radix_tree.cc:
```diff
@@ -353,7 +363,12 @@ TagGptr RadixTree::put(const char *key, const size_t key_size, Gptr value, Updat
             while(intermediate_node_ptr==0 && (cnt--)>0)
                 intermediate_node_ptr = heap->Alloc(sizeof(Node));
-            assert(intermediate_node_ptr.IsValid());
+			if(!intermediate_node_ptr.IsValid()){
```
Please revisit alignment for all the files.

gautham-bhat-k committed:
Addressing the first set of comments
1] Fixing indentation
2] Adding errorcode check for heap resize

gautham-bhat-k committed:
1] Added missing indentation fix
2] changed returning error code to assertion in radix_tree.cc after heap resize
3] modified error code in FindOrCreate
